### ISSUE TITLE
[Feature] Is matter support OOB(Out Of Bound) commission process?

### ISSUE THREAD

**Issue by dipVyas92** [project-chip/connectedhomeip#28516]
### Feature description

Hi,

We have one query regarding the commission process for matter devices over BLE.

We want to try commission matter end device over the BLE using the OOB(Out Of Bound) method.

OOB method means:- Commission tool has all the matter dataset details. Using that commissioning tool will write all datasets to the matter end device and the device will read all datasets and the device will connect to the matter network over an openthread.

Based on this feature the end device does not need to be in range with any IoT gateway or Matter Host.

Can Matter end device support this kind of OOB method to commission?

### Platform

k32w

### Platform Version(s)

Matter-v1.1

**Comment by dipVyas92** [comment]
@bzbarsky-apple can you please help me on this query?

**Comment by bzbarsky-apple** [comment]
@dipVyas92 I did look at this issue, but I don't understand what you are asking, exactly.  If you have some method of writing networking configuration to a device not using Matter, then that devices advertises commissionable via DNS-SD on that network, then commissioners can commission it.  If that's not the question, what is the question?

**Comment by dipVyas92** [comment]
Dear @bzbarsky-apple,

Thank you for getting back to me.

I want to use commissioning method to ensure the best possible results for end device. I suggest utilizing a mobile device as commissioning tool, which will seamlessly send all necessary Matter connection data to matter device. This will allow it to connect to OpenThread without any interference from the OpenThread Host, ensuring a smooth and reliable experience.

Please let me know if need more information regarding this.

I have attached two images for more clarification 

![BAEPhase_CSMIM-Page-67 drawio](https://github.com/project-chip/connectedhomeip/assets/131951062/fe98c336-90df-45be-a5ab-7beaa0dbbe36)

![BAEPhase_CSMIM-Page-68 drawio](https://github.com/project-chip/connectedhomeip/assets/131951062/212a9fa3-eecf-4e33-8bb3-0637b1aaf3a1)

Thanks in advance.

**Comment by bzbarsky-apple** [comment]
@dipVyas92 Yes, that's exactly how Matter commissioning works: you connect to the devices over BLE and provision them with the Thread credentials.

**Comment by dipVyas92** [comment]
@bzbarsky-apple 

Thanks for the response

Is it necessary for the end device to be within range of the IoT gateway or Openthread Host device for this commissioning process?

am I understanding correctly?

**Comment by bzbarsky-apple** [comment]
> Is it necessary for the end device to be within range of the IoT gateway or Openthread Host device for this commissioning process?

Well, commissioning is expected to put it on the Thread network _and then contact it over the Thread network to verify that worked_, so yes.

**Comment by pramodhrachuri** [comment]
Sorry for hijacking an ancient issue. Let me know if I should create a new one to discuss this.

1
Do the commissioner (like a mobile phone or a handheld terminal) and the commissionee (like a smart switch) have to communicate over BLE? Is there a provision for a cabled approach? Maybe over UART? I know this might be insecure, but in my use case, the commissioner is trustworthy, and the physical presence guarantees enough safety.


2  
> then contact it over the Thread network to verify that worked

Who is contacting whom here? Is the commissioner contacting the commissionee?
Or does the commissioner need to contact the administrator (like a Home Assistant device) over WiFi to check if the commissionee has joined?

From what I read, the thread protocol is mesh-based and thus can have a higher range (using multiple hops) than single WiFi. So, the verification through the thread will be much more convenient.

**Comment by bzbarsky-apple** [comment]
> Is there a provision for a cabled approach?

Not really, apart from an IP network over that cable (but then the commissioner might not realize it needs to put the device on some other IP network).

> Is the commissioner contacting the commissionee?

Yes.  The commissioner does a DNS-SD lookup for the commissionee, discovers its IP/port, and sends it a CommissioningComplete command.  None of this depends on Thread or WiFi per se, except as implementation details: as long as IP packets can be routed from the commissioner to the commissionee and back, and as long as the DNS-SD bits work correctly, it will work.

**Comment by pramodhrachuri** [comment]
Thanks for the response. I have a follow-up question.

Openthread's documentation says, "Only the Thread Network Key is needed to attach to a network." [[reference](https://openthread.io/reference/cli/commands#dataset_set_activepending:~:text=Only%20the%20Thread%20Network%20Key%20is%20needed%20to%20attach%20to%20a%20network.)]
Therefore, I can code a device (such as an ESP32c6) to receive the network key (+ other parts of the dataset) via UART and initiate the attachment process to join a Thread network. Correct?
Now that the device is on the Thread network and DNS-SD has started, we can initiate the matter commissioning process (like explained [here](https://docs.silabs.com/matter/1.0.3/matter-overview-guides/commissioning)). 

I'm thinking of two different paths after this.
1. If the commissioner (like a mobile) is already in the thread network, it should be able to handle the onboarding. Correct? Or does it still need a Bluetooth communication link?
2. Once the device is attached to Thread, can software like Home Assistant handle the commissioning process? This way, we can eliminate the need for a separate commissioner device.

Please correct me if any of my assumptions are wrong or my hypotheses are incorrect.

**Comment by bzbarsky-apple** [comment]
If the commissioner has IP connectivity to the device, then the commissioner can just perform commissioning over the IP network.  I think that covers both your questions.
